src/python/pants/option/config.py

```diff
@@ -154,8 +154,6 @@ def _determine_seed_values(*, seed_values: Optional[SeedValues] = None) -> Dict[
             "buildroot": buildroot,
             "homedir": os.path.expanduser("~"),
             "user": getpass.getuser(),
-            "pants_bootstrapdir": get_pants_cachedir(),
```

Comment:
We've been warning that these are unused for a while, so now that they are deprecated I assume we can simply not allow them to be used as seed values (IIRC this was why we had this "future deprecation" idiom).

Reply:
cc @Eric-Arellano ^

Reply:
Well, so, I updated the help message but never officially deprecated :/ I couldn't figure out how to without causing a deprecation warning for everyone due to this code here.

I think we need to revert removing this and the options.

Reply:
Well we have to remove them somehow... For now to get the release out I will punt.

Reply:
They are properly deprecated now, what will fail without a deprecation warning is any attempt to use them as seed values. And maybe that's OK? We never documented their existence as seed values, and I'm not sure we intended to commit to those being public API? I'd be surprised if anyone at all hits a problem with this.

Reply:
Yea, I'm fine with that.

Reply:
@Eric-Arellano , does that make sense? I don't see how we can deprecate the use of these as seed values without doing some non-trivial amount of disruptive work. And since they aren't really documented...

Reply:
Yes, this does sound sensible. Thank you for figuring this out. 

It'd be helpful to extract out the new deprecation into a dedicated PR so that it shows up in the changelog. It increases the likelihood someone realizes we did this. Probably fine if you want to keep here, though.